source/ast/expressions/AssertionExpr.cpp

```diff
@@ -523,22 +625,44 @@ void SimpleAssertionExpr::requireSequence(const ASTContext& context, DiagCode co
     }
 }
 
-bool SimpleAssertionExpr::admitsEmptyImpl() const {
+bitmask<NondegeneracyStatus> SimpleAssertionExpr::checkNondegeneracyImpl() const {
+    if (expr.kind == ExpressionKind::AssertionInstance) {
+        auto& aie = expr.as<AssertionInstanceExpression>();
+        if (aie.type->isSequenceType())
+            return aie.body.checkNondegeneracy();
+    }
+
+    bitmask<NondegeneracyStatus> res = NondegeneracyStatus::None;
+
+    // If simple sequence expression can be evaluated to `false` (`0`)
+    // manually then it should be reported as sequence
+    // with no possible match
+    res = (isNullExpr) ? res | NondegeneracyStatus::AdmitsNoMatch : res;
+
     if (repetition) {
         auto result = repetition->admitsEmpty();
-        if (result == SequenceRepetition::AdmitsEmpty::Yes)
-            return true;
-        if (result == SequenceRepetition::AdmitsEmpty::No)
-            return false;
+        res = (result == SequenceRepetition::AdmitsEmpty::Yes)
+                  ? res | NondegeneracyStatus::AdmitsEmpty
+                  : res;
+        if (result == SequenceRepetition::AdmitsEmpty::Only)
+            res = res | NondegeneracyStatus::AcceptsOnlyEmpty | NondegeneracyStatus::AdmitsEmpty;
     }
 
+    return res;
+}
+
+std::optional<SequenceRange> SimpleAssertionExpr::computeSequenceLengthImpl() const {
     if (expr.kind == ExpressionKind::AssertionInstance) {
         auto& aie = expr.as<AssertionInstanceExpression>();
         if (aie.type->isSequenceType())
-            return aie.body.admitsEmpty();
+            return aie.body.computeSequenceLength();
```

Comment:
Same comment here about skipping the repetition on this expression, which seems like it should affect the computed length.

Reply:
fixed